File: C/Grace.c
#include <stdio.h>
#include <fcntl.h>

/* Comment */
#define STRING char *str="#include <stdio.h>%1$c#include <fcntl.h>%1$c%1$c/* Comment */%1$c#define STRING char *str=%2$c%3$s%2$c%1$c#define PRINTF_PARAMS fd, str, 10, 34, str%1$c#define GRACE_PROGRAM int main(void){STRING; FILE *fd = fopen(%2$cGrace_kid.c%2$c, %2$cw%2$c); fprintf(PRINTF_PARAMS); fclose(fd);}%1$c%1$cGRACE_PROGRAM%1$c"
#define PRINTF_PARAMS fd, str, 10, 34, str
#define GRACE_PROGRAM int main(void){STRING; FILE *fd = fopen("Grace_kid.c", "w"); fprintf(PRINTF_PARAMS); fclose(fd);}

GRACE_PROGRAM

File: Python/Grace.py
#Comment
STRING = "#Comment%(nl)cSTRING = %(qu)c%(string)s%(qu)c%(nl)cWRITE_MAP = {'nl': 10, 'qu': 34, 'string': STRING}%(nl)cGRACE_PROGRAM = exec('file = open(%(qu)cGrace_kid.py%(qu)c, %(qu)cw+%(qu)c); file.write(STRING %% WRITE_MAP); file.close()')%(nl)c%(nl)cGRACE_PROGRAM%(nl)c"
WRITE_MAP = {'nl': 10, 'qu': 34, 'string': STRING}
GRACE_PROGRAM = exec('file = open("Grace_kid.py", "w+"); file.write(STRING % WRITE_MAP); file.close()')

GRACE_PROGRAM
